src/transforms/groupby.js

```diff
@@ -35,6 +35,35 @@ exports.attributes = {
             'with `x` [1, 3] and one trace with `x` [2, 4].'
         ].join(' ')
     },
+    nameformat: {
```

Comment:
Bringing in @alexcjohnson @cldougl @chriddyp @cpsievert @monfera what do you think of this syntax?

A lot of existing attributes could benefit form _special formatting_ character like this one. For example,

```js
trace = {
  y: [1, 2, 1]
  marker: {
    size: [20, 30, 10]
  },
  hovertext: '%marker.size'
}
```

would be a nice shortcut to show each point's `marker.size` value on hover.